File: AlgorithmsPackPythonSolutions/1200. Minimum Absolute Difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        sz = len(arr)
        
        minDiff = float('inf')
        for index in range(1, sz):
            minDiff = min(minDiff, arr[index] - arr[index-1])
        
        result = []
        for index in range(1, sz):
            if arr[index] - arr[index-1] == minDiff:
                result.append([arr[index-1], arr[index]])
        
        return result

File: AlgorithmsPackJavaSolutions/1200. Minimum Absolute Difference.java
public class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort(arr);
        int sz = arr.length;
        
        int minDiff = Integer.MAX_VALUE;
        for (int index = 1; index < sz; index++) {
            minDiff = Math.min(minDiff, arr[index] - arr[index-1]);
        }
        
        List<List<Integer>> result = new ArrayList<>();
        for (int index = 1; index < sz; index++) {
            if (arr[index] - arr[index-1] == minDiff) {
                List<Integer> pair = new ArrayList<>();
                pair.add(arr[index-1]);
                pair.add(arr[index]);
                result.add(pair);
            }
        }
        
        return result;
    }
}
